internal/graph/parser/parser.go

```diff
@@ -88,6 +92,15 @@ func parseObject(field map[string]interface{}, schema *spec.Schema, path, expect
 		return nil, fmt.Errorf("expected object type or AdditionalProperties allowed for path %s, got %v", path, field)
 	}
 
+	// Look for vendor schema extensions first
+	if len(schema.VendorExtensible.Extensions) > 0 {
+		// If the schema has the x-kubernetes-preserve-unknown-fields extension, we should not
+		// parse the object and return an empty list of expressions.
+		if enabled, ok := schema.VendorExtensible.Extensions[xKubernetesPreserveUnknownFields]; ok && enabled.(bool) {
+			return nil, nil
```

Comment:
Instead of ignoring all CEL expressions inside RawExtension fields, we could just turtle it down using the schemaless parser https://github.com/awslabs/kro/blob/main/internal/graph/parser/schemaless.go#L23-L25